docs/README.md

```diff
@@ -1,20 +1,201 @@
+
+### The `do_conf` function
+
+As one might expect, there are many values which a DAQ module may rely on to perform its calculations when in the running state that ideally should be settable during the `conf` transition. The typical technique is to have some member data which in the DAQ module constructor intentionally gets initialized either to zero or to implausible values (e.g. `m_calibration_scale_factor(-1)`, `m_num_total_warnings(0)`) and then to set them properly during the `config` transition. You'll see in the code below that the type of the data instance `data` which gets extracted from the JSON is `mydaqmodule::Conf`, and then `data` is used to set the member(s). 
+```
+void MyDaqModule::do_conf(const data_t& conf_data)
+{
+  auto data = conf_data.get<mydaqmodule::Conf>();
+
+  m_calibration_scale_factor = data.calibration_scale_factor;
+  // ...and then set the other members which take per-configuration values...
+}
+```
+This of course raises the question: what _is_ `mydaqmodule::Conf`? It's a `struct`, but rather than being manually written the code for it is generated by the DUNE DAQ build system itself, using a `jsonnet` file as input. It's in the `jsonnet` file that the logical contents of the struct are defined; an example of this type of file can be found [here](https://github.com/DUNE-DAQ/listrev/blob/develop/schema/listrev/randomdatalistgenerator.jsonnet). Further details are beyond the scope of appfwk documentation and are instead covered in the [daq-cmake documentation](https://dune-daq-sw.readthedocs.io/en/latest/packages/daq-cmake/).
+
+### The `do_start` function
+
+Most DAQ modules are designed to loop over some sort of repeated action when the DAQ enters the running state, and it's in the `do_start` function that this repeated action begins. A very common idiom for the `do_start` function is, "Set an atomic boolean stating that we're now in the running state, and then start one or more threads which perform actions in loops which they break out of if they see that the atomic boolean indicates we're no longer in the running state". 
+
+While it's of course possible to accomplish this using the existing concurrency facilities provided by the C++ Standard Library, the appfwk package itself provides a class, `ThreadHelper`, which makes this easier. `ThreadHelper` is covered in detail [here](https://dune-daq-sw.readthedocs.io/en/latest/packages/appfwk/ThreadHelper-Usage-Notes/); when in use the `do_start` function can be as simple as follows:
+```
+void MyDaqModule::do_start(const data_t& /*args*/) {
+    m_thread.start_working_thread();  // m_thread is an `appfwk::ThreadHelper` member of MyDaqModule
+}
+```
```

Comment:
Here I would add that the ThreadHeper does not only help the implementor, but also the user (in the sense of shifter). Since the threadhelper makes renaming the thread easy, it makes the control of the operation easier. 
I would also add that because of that using ThreadHelper is preferred over STL solutions.
STL solution should be considered only if more advanced features other than those offered by ThreadHelper are required, e.g. future, promise, deferred launch, etc.